### DISCUSSION TITLE
Lesson-7 | Part-6 | Invalid type for argument in function call.

### DISCUSSION THREAD

**Discussion by vigneshbalu96** [Cyfrin/foundry-full-course-cu#1991] [General]
while refactoring the code it gives me this error of [Invalid Type]

```solidity
function fund() public payable {
    require(msg.value.getConversionRate(s_priceFeed) >= MINIMUM_USD, "You need to spend more ETH!");
    // require(PriceConverter.getConversionRate(msg.value) >= MINIMUM_USD, "You need to spend more ETH!");
    addressToAmountFunded[msg.sender] += msg.value;
    funders.push(msg.sender);
}
```

while hovering over the `s_priceFeed` in

> require(msg.value.getConversionRate(s_priceFeed)`

it shows

`
Invalid type for argument in function call. Invalid implicit conversion from contract AggregatorV3Interface to contract AggregatorV3Interface requested.solidity(9553)
AggregatorV3Interface private s_priceFeed
`

**Comment by EngrPips** [comment]
Hello @vigneshbalu96, visit this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/1948#discussion-6902122. It is about the same issue you are having.

**Reply by vigneshbalu96** [reply]
Thank You @EngrPips

**Reply by EngrPips** [reply]
You are welcome, Friend.
